enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -280,20 +272,96 @@ does the following:
-support, which means that Istio watches Gateway and HTTPRoute objects and
-configures Envoy accordingly.  Configuring a load balancer and DNS enables
-clients to connect through Envoy to backend applications as specified by the
-Gateway and HTTPRoute objects.
+When OSSM is installed, it deploys the Istio control-plane.  The Ingress
+Operator configures the ServiceMeshControlPlane to enable Gateway API support,
+which means that Istio watches Gateway and HTTPRoute objects and configures
+Envoy accordingly.  The Ingress Operator also configures the
+ServiceMeshControlPlane to enable Istio's "[Automated Deployment](https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/#automated-deployment)" feature,
+which means that Istio automatically creates an Envoy proxy deployment and a
+LoadBalancer-type service for each Gateway.  The Ingress Operator then
+configures DNS for each such service (see [New Controller to Manage DNS Records
+for Gateway Listeners](#new-controller-to-manage-dns-records-for-gateway-listeners)).  With the Ingress Operator and Service Mesh Operator
+together configuring a load balancer and DNS, this enables clients to connect
+through Envoy to backend applications as specified by the Gateway and HTTPRoute
+objects.
+
+Following is the ServiceMeshControlPlane CR that the Ingress Operator creates:
+
+```yaml
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: gwapi
+  namespace: openshift-ingress
+spec:
+  profiles:
+  - gateway-controller
+  proxy:
+    accessLogging:
+      envoyService:
+        enabled: true
+      file:
+        name: /dev/stdout
+  runtime:
+    components:
+      pilot:
+        container:
+          env:
+            PILOT_GATEWAY_API_CONTROLLER_NAME: openshift.io/gateway-controller
+  version: v2.4
+  techPreview:         # [1]
+    controlPlaneMode: ClusterScoped
+```
+[1]: Specifying `techPreview.controlPlaneMode: ClusterScoped` is currently required even when specifying `spec.profiles: gateway-controller` due to a known issue in OSSM.
+
+This SMCP uses OSSM's built-in "gateway-controller" profile to enable Gateway
+API.  Specifying this profile is equivalent to specifying the following
+configuration on the SMCP:
+
+```yaml
+spec:
+  techPreview:
+    controlPlaneMode: ClusterScoped
+    gatewayAPI:
+      enabled: true
+      controllerMode: true
+  addons:
+    grafana:
+      enabled: false
+    jaeger:
+      install: {}
+      name: jaeger
+    kiali:
+      enabled: false
+    prometheus:
+      enabled: false
+  gateways:
+    egress:
+      enabled: false
+    ingress:
+      enabled: true
+      ingress: true
+  policy:
+    type: Istiod
+  security:
+    manageNetworkPolicy: false
+  tracing:
+    type: None
+```
+
+The above configuration enables Gateway API with a cluster-scoped control-plane.
+This means that OSSM watches Gateway API resources in all namespaces.  The above
+configuration also enables ingress and egress proxies for service mesh; a
```

Comment:
Are you sure you want to enable ingress and egress proxies for SM in dev preview? It would mean alot more testing.

Reply:
It would probably be reasonable to do a test to make sure the mesh functionality works with an example application.  More than that isn't necessary for dev preview IMO.